packages/edit-site/src/components/page-pages/index.js

```diff
@@ -352,13 +355,24 @@ export default function PagePages() {
 	const permanentlyDeletePostAction = usePermanentlyDeletePostAction();
 	const restorePostAction = useRestorePostAction();
 	const editPostAction = useEditPostAction();
+	const customizedEditPostAction = {
+		...editPostAction,
+		callback: ( posts ) => {
+			const post = posts[ 0 ];
+			history.push( {
+				postId: post.id,
+				postType: post.type,
+				canvas: 'edit',
+			} );
```

Comment:
I believe we have some "commands" that either navigate using the url or use history depending on whether we detect that we're in the site editor. Can't we do the same thing within the shared action and avoid overriding it entirely?